### PR TITLE
Correct documention of GET GroupResources

### LINKED ISSUE #27694: auth/admin/realms/{realm-name}/groups?briefRepresentation=false returns empty subgroups.
### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

core

### Describe the bug

Hi Team,

We have been migrating to KC 23.5 and observed that the /groups functionality doesn't return subgroups anymore and the field is empty. 
Can you suggest an alternative way/fix to handle the same? 

### Version

23.5

### Regression

- [ ] The issue is a regression

### Expected behavior

The groups should be returned with subgroups populated.

### Actual behavior

Subgroup field is empty

### How to Reproduce?

https://{keycloak-host}:port/auth/admin/realms/{realm-name}/groups?briefRepresentation=false

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by RichieB2B**
Closes #27694

Fixes erroneous statement added by #34128

### CODE REVIEW

File: services/src/main/java/org/keycloak/services/resources/admin/GroupsResource.java [hunk 1]
```diff
@@ -77,14 +77,14 @@ public GroupsResource(RealmModel realm, KeycloakSession session, AdminPermission
 
     /**
      * Get group hierarchy.  Only {@code name} and {@code id} are returned.  {@code subGroups} are only returned when using the {@code search} or {@code q} parameter.
-     * If these parameters are empty, the top-level groups are returned without `{@code subGroups} being filled.
+     * If these parameters are empty, the top-level groups are returned with `{@code subGroups} being filled.
```
Comment (ahus1) [inline services/src/main/java/org/keycloak/services/resources/admin/GroupsResource.java:80]:
```suggestion
     * If none of these parameters is provided, the top-level groups are returned without `{@code subGroups} being filled.
```

File: services/src/main/java/org/keycloak/services/resources/admin/GroupsResource.java [hunk 2]
```diff
@@ -77,14 +77,14 @@ public GroupsResource(RealmModel realm, KeycloakSession session, AdminPermission
...
      * @return
      */
     @GET
     @NoCache
     @Produces(MediaType.APPLICATION_JSON)
     @Tag(name = KeycloakOpenAPI.Admin.Tags.GROUPS)
-    @Operation( summary = "Get group hierarchy.  Only `name` and `id` are returned.  `subGroups` are only returned when using the `search` or `q` parameter. If these parameters are empty, the top-level groups are returned without `subGroups` being filled.")
+    @Operation( summary = "Get group hierarchy.  Only `name` and `id` are returned.  `subGroups` are only returned when using the `search` or `q` parameter. If these parameters are empty, the top-level groups are returned with `subGroups` being filled.")
```
Comment (ahus1) [inline services/src/main/java/org/keycloak/services/resources/admin/GroupsResource.java:87]:
```suggestion
    @Operation( summary = "Get group hierarchy.  Only `name` and `id` are returned.  `subGroups` are only returned when using the `search` or `q` parameter. If none of these parameters is, the top-level groups are returned without `subGroups` being filled.")
```

**Review by ahus1 — CHANGES_REQUESTED** [review]
In my original comment I mixed up `empty` with `not provided`, so this is what had in mind originally. Please let me know if this makes sense.

**Comment by RichieB2B** [comment]
I totally agree but please remember the whole reason for this documentation change is the strange behaviour of `getGroups` since the 23.0 release. In #27694 this was described as intended behaviour but the breaking change has not been documented anywhere. This keeps confusing users of the API. If you revert the docs change, please also fix the bug. Otherwise we'll be back at square one.

**Review by ahus1 — APPROVED** [review]
As it is unclear when the original issue is handled, let's update the docs only. I'll re-open the original issue once this PR is merged.

**Pull request merged**